python/tvm/relay/testing/lstm.py

```diff
@@ -0,0 +1,78 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Implementation of a Long Short-Term Memory (LSTM) cell.
+
+Adapted from:
+https://gist.github.com/merrymercy/5eb24e3b019f84200645bd001e9caae9
+"""
+
+from tvm import relay
+from . import layers
+from .init import create_workload
+
+def lstm_cell(inputs, states, i2h_weight, h2h_weight,
+              i2h_bias, h2h_bias, num_hidden):
+    """Long-Short Term Memory (LSTM) network cell.
+
+    Parameters
+    ----------
+    inputs : relay.Expr
+        Input data to LSTM cell.
+
+    states : Tuple[relay.Expr, relay.Expr]
+        State for LSTM cell. Should be of shape (batch size, num_hidden)
+
+    i2h_weight : relay.Expr
+        i2h weight to LSTM cell.
+
+    h2h_weight : relay.Expr
+        h2h weight to LSTM cell.
+
+    i2h_bias : relay.Expr
+        i2h bias to LSTM cell.
+
+    h2h_bias : relay.Expr
+        h2h bias to LSTM cell.
+
+    num_hidden : int
+        Number of units in output symbol.
+
+    Returns
+    -------
+    result : Tuple[relay.Expr, Tuple[relay.Expr, relay.Expr]]
+        The result.
+    """
+
+    i2h = layers.dense_add_bias(data=inputs, weight=i2h_weight,

```

Comment:
Weight and bias should be left out. They will be created automatically by `dense_add_bias`

Reply:
Interesting